backend/dataall/core/permissions/services/tenant_policy_service.py

```diff
@@ -25,9 +18,130 @@ def is_tenant_admin(groups: [str]):
 
         return False
 
+    @staticmethod
+    def validate_admin_access(username, groups, action):
+        if not TenantPolicyValidationService.is_tenant_admin(groups):
+            raise exceptions.UnauthorizedOperation(
+                action=action,
+                message=f'User: {username} is not allowed to manage tenant permissions',
+            )
+
+    @staticmethod
+    def validate_find_tenant_policy(group_uri, tenant_name):
+        if not group_uri:
+            raise exceptions.RequiredParameter(param_name='group_uri')
+        if not tenant_name:
+            raise exceptions.RequiredParameter(param_name='tenant_name')
+
+    @staticmethod
+    def validate_attach_tenant_policy(group, permissions, tenant_name):
+        if not group:
+            raise exceptions.RequiredParameter(param_name='group')
+        if not permissions:
+            raise exceptions.RequiredParameter(param_name='permissions')
+        if not tenant_name:
+            raise exceptions.RequiredParameter(param_name='tenant_name')
+
+    @staticmethod
+    def validate_save_tenant_policy(group, tenant_name):
+        if not group:
+            raise exceptions.RequiredParameter(param_name='group')
+        if not tenant_name:
+            raise exceptions.RequiredParameter(param_name='tenant_name')
+
+    @staticmethod
+    def validate_add_permission_to_tenant_policy_params(group, permissions, policy, tenant_name):
+        if not group:
+            raise exceptions.RequiredParameter(param_name='group')
+        TenantPolicyValidationService.validate_add_permissions_params(permissions, policy, tenant_name)
+
+    @staticmethod
+    def validate_add_permissions_params(permissions, policy, tenant_name):
+        if not permissions:
+            raise exceptions.RequiredParameter(param_name='permissions')
+        if not tenant_name:
+            raise exceptions.RequiredParameter(param_name='tenant_name')
+        if not policy:
+            raise exceptions.RequiredParameter(param_name='policy')
+
```

Comment:
The name of this function is too generic for being used just once in a very particular case (action: `UPDATE_TENANT_TEAM_PERMISSIONS`) Either we make the function more generic or we rename it to be more specific